x-pack/test/functional/services/observability/users.ts

```diff
@@ -24,7 +26,8 @@ export function ObservabilityUsersProvider({ getPageObject, getService }: FtrPro
    */
   const setTestUserRole = async (roleDefinition: CreateRolePayload) => {
     // return to neutral grounds to avoid running into permission problems on reload
-    await commonPageObject.navigateToActualUrl('kibana');
+    await commonPageObject.navigateToActualUrl('home');
```

Comment:
Because of a race condition, sometimes the redirect happens quickly, and as a result, `/home` path is compared to `/kibana`. I removed the extra redirect and added a check after this line to ensure the page is loaded properly before moving to the next step.

This should fix the failure in this part which prevented role creation after this step and as a result, having a not found when deleting the role (`AxiosError: Request failed with status code 404`)